src/components/QuranReader/TranslationView/ShareVerseButton.tsx

```diff
@@ -0,0 +1,44 @@
+import clipboardCopy from 'clipboard-copy';
+import useTranslation from 'next-translate/useTranslation';
+
+import ShareIcon from '../../../../public/icons/share.svg';
+
+import Button, { ButtonSize, ButtonType } from 'src/components/dls/Button/Button';
+import { ToastStatus, useToast } from 'src/components/dls/Toast/Toast';
+import { getWindowOrigin } from 'src/utils/url';
+import { getVerseAndChapterNumbersFromKey } from 'src/utils/verse';
+
+type ShareVerseButtonProps = {
+  verseKey: string;
+};
+
+export const onShareClicked = (verseKey, callback: () => void) => {
+  const origin = getWindowOrigin();
+  const [chapter, verse] = getVerseAndChapterNumbersFromKey(verseKey);
+  if (origin) {
+    clipboardCopy(`${origin}/${chapter}/${verse}`).then(callback);
```

Comment:
This is an example of a place where we should be doing the logging. You can sync with Osama on how it should be done